### ISSUE TITLE
Execute concurrent mutations

### ISSUE THREAD

**Issue by clagraff** [clagraff/devoid#6]
Intents should be handled concurrently, withouth locking entities (maybe they get a copy of the structs they need?).

When mutations from intents are created, they should only change 1 entity per mutation.
Then mutations should operate on the pointers of entity structs, concurrently, locking the one struct they need.

This is essentially to revert: https://github.com/clagraff/devoid/pull/4/files#diff-91bbeda7eb98a7adc57b9e47e2cf5c2bR171

**Comment by clagraff** [comment]
Intents should be read-only, requesting entities by ID or position or whatever, and get a copy. That copy though, should be locked if it is currently being written to.
Then do normal  intent stuff.

For mutators, each individual mutator should operate on only ONE entity at a time, and lock that entity until it is no longer needed.
Then, both intents and mutators can happen in parallel

**Comment by clagraff** [comment]
So maybe instead of providing the State object, we expose an interface for requesting a copy of an enitity by ID or position, which blocks if that entity is currently being written to.

**Comment by clagraff** [comment]
Done
